Classes/ContentDefender/Xclasses/DatamapHook.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace B13\Container\ContentDefender\Xclasses;
+
+/*
+ * This file is part of TYPO3 CMS-based extension "container" by b13.
+ *
+ * It is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, either version 2
+ * of the License, or any later version.
+ */
+
+use B13\Container\ContentDefender\ContainerColumnConfigurationService;
+use IchHabRecht\ContentDefender\Hooks\DatamapDataHandlerHook;
+use IchHabRecht\ContentDefender\Repository\ContentRepository;
+use TYPO3\CMS\Core\DataHandling\DataHandler;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Core\Utility\MathUtility;
+
+class DatamapHook extends DatamapDataHandlerHook
+{
+
+    /**
+     * @var ContainerColumnConfigurationService
+     */
+    protected $containerColumnConfigurationService;
+
+    protected $mapping = [];
+
+    public function __construct(
+        ContentRepository $contentRepository = null,
+        ContainerColumnConfigurationService $containerColumnConfigurationService = null
+    ) {
+        $this->containerColumnConfigurationService = $containerColumnConfigurationService ?? GeneralUtility::makeInstance(ContainerColumnConfigurationService::class);
+        parent::__construct($contentRepository);
+    }
+
+    /**
+     * @param DataHandler $dataHandler
+     */
+    public function processDatamap_beforeStart(DataHandler $dataHandler): void
+    {
+        if (is_array($dataHandler->datamap['tt_content'] ?? null)) {
```

Comment:
I needed a $dataHandler->isOuterMostInstance() check here additionally, otherwise the following happens:

scenario:
- outer container element with content_defender allowed but no max items
- inner container element with content_defender allowed and max items = 1
- content element inside that container (filling the col with max items=1)
- setup new outer container + inner container
- copy innermost content element (the one from the max items =1 col)
- try to paste into "free" inner column of second element === error

Internally, the datahandler tries to create the copied (NEW123) element in the original column and wants to update the values after that - inserting into the original column fails because the max item restriction here prevents the insert round and doesn't know anything about the update to a different parent/colPos later.

Reply:
Hi @susannemoog , thanks for testing and Feedback.

can you try this patch:
```
diff --git a/Classes/Hooks/Datahandler/CommandMapBeforeStartHook.php b/Classes/Hooks/Datahandler/CommandMapBeforeStartHook.php
index cf92b08..96ee205 100644
--- a/Classes/Hooks/Datahandler/CommandMapBeforeStartHook.php
+++ b/Classes/Hooks/Datahandler/CommandMapBeforeStartHook.php
@@ -150,6 +150,9 @@ class CommandMapBeforeStartHook
                         isset($cmd['update']['colPos'])
                     ) {
                         $cmd['update'] = $this->dataFromContainerIdColPos($cmd['update']);
+                        if (isset($cmd['update']['tx_container_parent']) && $cmd['update']['tx_container_parent'] > 0 && $cmd['target'] > 0) {
+                            $cmd['target'] = -$cmd['update']['tx_container_parent'];
+                        }
                     }
                 }
             }
```

currently i work also on wrong sortings [#149](https://github.com/b13/container/issues/149) and i think pasting an element into an container at first possition should change the target, so

```        $cmdmap = [
            'tt_content' => [
                42 => [
                    'copy' => [
                        'action' => 'paste',
                        'target' => 1,
                        'update' => [
                            'colPos' => '41-201',
                            'sys_language_uid' => '0'
                        ]
                    ]
                ]
            ]
        ];
```
(meens pasting CE 42 into Container 41 at colPos 201 (on page 1))
becomes
```
        $cmdmap = [
            'tt_content' => [
                42 => [
                    'copy' => [
                        'action' => 'paste',
                        'target' => -41,
                        'update' => [
                            'colPos' => '41-201',
                            'sys_language_uid' => '0'
                        ]
                    ]
                ]
            ]
        ];
```
so the element is paste after container 41

(this works for me for your described bug, if i understood correct)

Reply:
argh, also other scenarios are affected, needs move improvments, 
work on it

Reply:
As discussed in Slack, with this change at least the maxItems scenario works - tested copying the single element, the complete container with sub elements and copying the container into a container.

Reply:
well... too fast... https://www.awesomescreenshot.com/video/6189182?key=60c3f4a7211022f5a5f071e0a1a2bd73 broke it again.